modules/developer_manual/examples/scripts/backport.sh

```diff
@@ -1,16 +1,39 @@
 #!/bin/bash
 
-if ! [ -x "$(command -v jq)" ]; then
-  echo 'Error: jq is not installed.' >&2
-  echo 'Please install package "jq" before using this script'
-  exit 1
-fi
-
-if ! [ -x "$(command -v curl)" ]; then
-  echo 'Error: curl is not installed.' >&2
-  echo 'Please install package "curl" before using this script'
-  exit 1
-fi
+set -o pipefail
+set -o nounset
+
+function printCommitList() {
+    local commitList="$1"
+
+    # TODO: Might be able to refactor this to be somewhat simpler. Maybe.
+    commitCount=$(echo "$commitList" | grep '' | wc -l | tr -d ' ')
+    if [[ $commitCount -ne 0 ]]; then
+        if [[ $commitCount -eq 1 ]]; then
+            echo "$commitCount commit being cherry picked:"
+        else
+            echo "$commitCount commits being cherry picked:"
+        fi
+
+        for i in $commitList; do 
+            printf "* %s\n" $i 
+        done
+    fi
+}
+
+function checkBinaryIsAvailable() {
+    local binary="$1"
+    if ! [ -x "$(command -v ${binary})" ]; then
+        echo "Error: ${binary} is not installed." >&2
+        echo "Please install package "${binary}" before using this script"
+        exit 1
+    fi
+}
+
+requiredBinaries=('jq' 'curl' 'git' 'grep' 'perl')
```

Comment:
This could likely be simplified with `xargs`, such as with `printf "%s\n" "${requiredBinaries[@]}" | xargs -I % sh -c 'command -v %'`.